### ISSUE TITLE
[taskcluster-web] Add ability to search for secrets

### ISSUE THREAD

**Issue by mitchhentges** [taskcluster/taskcluster#175]
If I want to jump to existing secrets, the new UI doesn't allow me to search for them.

**Potential solution:**
Like the other pages, show a search in the top-right that will filter the results

**Comment by helfi92** [comment]
STR:

1. Navigate to https://taskcluster-web.netlify.com/secrets

Expected: A search bar is shown top right allowing a user to search a secret after hitting on the enter key, similar to how we're able to search clients in https://taskcluster-web.netlify.com/auth/clients.

Actual: No search bar is shown.

You can see how it's being done one some of the site other views like:
* https://taskcluster-web.netlify.com/auth/roles ([code](https://github.com/taskcluster/taskcluster/blob/master/ui/src/views/Roles/ViewRoles/index.jsx))
* https://taskcluster-web.netlify.com/auth/clients ([code](https://github.com/taskcluster/taskcluster/blob/master/ui/src/views/Clients/ViewClients/index.jsx))

**Comment by twisted-sres** [comment]
Hey, can I take this up?

**Comment by helfi92** [comment]
Go for it :)

**Comment by twisted-sres** [comment]
Hey, I was trying to build taskcluster but I am getting this error
![screenshot from 2019-02-07 06-01-10](https://user-images.githubusercontent.com/36242614/52366449-c668bc80-2a6f-11e9-94b6-b8fe7f84bb80.png)

**Comment by mitchhentges** [comment]
Hmm, are you running docker?
If you `systemctl status docker.service`, it should say that it's running. If not, make sure it's started!

If it _is_ running, perhaps you have a permissions issue?
Try following the "Manage Docker as a non-root user" guide at the top of [this instructions page](https://docs.docker.com/install/linux/linux-postinstall/)

**Comment by twisted-sres** [comment]
![screenshot from 2019-02-07 06-21-28](https://user-images.githubusercontent.com/36242614/52367652-90790780-2a72-11e9-83de-cccd6442153c.png)
Now I am getting this

**Comment by mitchhentges** [comment]
Intesting, I'd recommend looking at the log file it says it created at `/tmp/taskcluster-builder-build/service-docs/yarn.log`, maybe it'll tell you what's going wrong

**Comment by helfi92** [comment]
@twisted-sres Why are you trying to build taskcluster?

To run the ui, clone the project, navigate to the ui directory, run `yarn` to install the dependencies and then `yarn start` to start the development server.

To run the server, from the root directory, run `yarn` to install the dependencies and then `yarn workspace taskcluster-web-server start`.

**Comment by twisted-sres** [comment]
Thanks @mitchhentges @helfi92 , I have got it running but it shows this
![screenshot from 2019-02-07 06-57-47](https://user-images.githubusercontent.com/36242614/52369943-c4a2f700-2a77-11e9-8377-fefc233ea496.png)

**Comment by helfi92** [comment]
Did you forget to create a `user-config.yml` file by any chance with `rootUrl` set to `https://taskcluster.net`? Instructions for the server can be found in https://github.com/taskcluster/taskcluster/tree/master/services/web-server#running-taskcluster-web-server-locally. If it's still not working, please show a screenshot of both the ui and server terminal output.

**Comment by vaniyst** [comment]
Is this bug fixed ?

**Comment by twisted-sres** [comment]
![screenshot from 2019-02-07 15-46-12](https://user-images.githubusercontent.com/36242614/52404642-821d0100-2aef-11e9-9467-a04ce7554801.png)
I've now created the file like this, and the ui shows the same result
![screenshot from 2019-02-07 15-48-22](https://user-images.githubusercontent.com/36242614/52404769-cb6d5080-2aef-11e9-81a8-b52e7dd240fe.png)

**Comment by helfi92** [comment]
Remove the semicolon from the rootUrl. It should be `"https://taskcluster.net"`, not `"https://taskcluster.net";`

**Comment by twisted-sres** [comment]
It still shows the same thing, user-config.yml file had to be in taskcluster/services/web-server only no?

**Comment by helfi92** [comment]
Yes, only web-server. Is it possible you've done some other changes? Is it possible you've changed `config.yml` by accident? Don't forget to restart the server after fixing `user-config.yml`. Maybe that would do it.

**Comment by twisted-sres** [comment]
![screenshot from 2019-02-08 22-15-03](https://user-images.githubusercontent.com/36242614/52492342-fc807a80-2bee-11e9-8093-82435a5930b1.png)
I tried doing it all, but it still shows the same error

**Comment by helfi92** [comment]
The screenshot you are showing is for the user interface. You still need the web-server component running in a separate terminal. Do you have this running?

**Comment by twisted-sres** [comment]
Yeah, I did that now, but the same results
![screenshot from 2019-02-09 01-01-52](https://user-images.githubusercontent.com/36242614/52501411-49238000-2c06-11e9-94e2-b721e4439d51.png)
![screenshot from 2019-02-09 01-01-30](https://user-images.githubusercontent.com/36242614/52501414-4a54ad00-2c06-11e9-87ee-1ac771d00c9f.png)
![screenshot from 2019-02-09 00-56-41](https://user-images.githubusercontent.com/36242614/52501415-4b85da00-2c06-11e9-8cbb-6464e8c3e2ab.png)

**Comment by helfi92** [comment]
Where did you create `user-config.yml`?

**Comment by twisted-sres** [comment]
Yeah, there only

**Comment by helfi92** [comment]
Can you try copy pasting the content of [this file](https://raw.githubusercontent.com/taskcluster/taskcluster/master/services/web-server/config.yml) in `config.yml`. I'm wondering if it got changed by accident. Can you also run the server with `DEBUG=* yarn start` and see if there's anything interesting outputted.

**Comment by twisted-sres** [comment]
@helfi92 Is it possible that any of this warning is causing the error?
![screenshot from 2019-02-09 15-02-57](https://user-images.githubusercontent.com/36242614/52519064-d3f29200-2c7b-11e9-89e2-0f3351ef4090.png)

**Comment by helfi92** [comment]
> @helfi92 Is it possible that any of this warning is causing the error?

No, that should be okay. Your issue is coming from the backend, not the front end. Could you try https://github.com/taskcluster/taskcluster/issues/175#issuecomment-461965534, please? Maybe a screenshot of the last couple of debug statements would help to understand what's going on :)

**Comment by helfi92** [comment]
How is it going @twisted-sres? Let me know if you had any questions.

**Comment by twisted-sres** [comment]
![screenshot from 2019-02-12 20-18-42](https://user-images.githubusercontent.com/36242614/52858794-2b4ba300-3151-11e9-9b61-79012cfe2e6e.png)
This shows up when I run server with DEBUG=* yarn start

**Comment by helfi92** [comment]
Is this the end of the output? There seems to be more stuff from what I can see in the screenshot.

**Comment by Aditya-Kolla** [comment]
If this is unassigned, can I work on this?

**Comment by helfi92** [comment]
Sure, go for it.

**Comment by Aditya-Kolla** [comment]
@helfi92  Do we need the defaultValue in secrets too? Like in clients? 
If yes, what would the defaultValue in this case be?

**Comment by helfi92** [comment]
> Do we need the defaultValue in secrets too? Like in clients?
If yes, what would the defaultValue in this case be?

Good question. Let's not have a `defaultValue` for now. We can always revisit later if we think it's necessary.

**Comment by Aditya-Kolla** [comment]
Alright :)

**Comment by Aditya-Kolla** [comment]
Hey, I made some progress into this. 
I've managed to make the search work but when I'm setting state and trying to refetch data the filter won't work.

![screenshot from 2019-02-18 23-27-49](https://user-images.githubusercontent.com/31780340/52969092-25fe8a80-33d5-11e9-98ff-2683e7f6fce7.png)


Any suggestions what I'm doing wrong?

One another thing I observed was that the main content was getting rendered twice which might affect the performance.(?) Not too sure about this.
